File: app/src/main/java/com/example/intentpractice/MainActivity.java
package com.example.intentpractice;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    Button add_btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        add_btn = findViewById(R.id.add_btn);



        add_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), SecondActivity.class)
                        .setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
                startActivity(intent);
            }
        });
    }


}

File: ClickPractice/app/src/main/java/com/amazing/android/clickpractice/MainActivity.kt
package com.amazing.android.clickpractice

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.widget.Button
import androidx.activity.viewModels
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import com.amazing.android.clickpractice.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private val viewModel: MainViewModel by viewModels()
    private var clickCount = 0;
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = DataBindingUtil.setContentView(this,R.layout.activity_main)

        //viewModel = ViewModelProvider(this).get(MainViewModel::class.java)

        //binding.lifecycleOwner = this

        binding.clickBtn.setOnClickListener {
            viewModel.incrementCounter()
        }

        binding.start.setOnClickListener {
            viewModel.checkTime()
        }

        binding.set.setOnClickListener {
            viewModel.set()
        }

        viewModel.counter.observe(this) { count ->
            binding.clickCount.text = count.toString()
        }

        viewModel.time.observe(this) { time ->
            binding.time.text = time.toString()
        }

    }

}